src/frontend/js/tinymce-plugins/custom_link_input/plugin.js

```diff
@@ -116,7 +116,7 @@ import { getCsrfToken } from "../../utils/csrf-token";
 
                 // Automatically update the text input to the url by default
                 data = api.getData();
-                if (!text_disabled && !url_changed_by_search && data.url != data.text && prev_link_url != data.url && (data.text != user_data.text || !data.text)) {
+                if (!text_disabled && !url_changed_by_search && data.url != data.text && prev_link_url != data.url && (data.text != user_data.text) && !data.text) {
```

Comment:
Probably the text should only be set to the url if the current text is equal to the previous url
 
This could simplify the condition a bit:

```suggestion
                if (!text_disabled && !url_changed_by_search && data.text == prev_link_url) {
```

Reply:
Thanks, I added your suggestion to the PR :)